File: .lmbrain-lite/batteria/fixture/ts-carrello/src/money.ts
/** Rounds to the nearest integer, halves away from zero: 2.5 -> 3, -2.5 -> -3. */
export function roundHalfUp(value: number): number {
  const rounded = Math.floor(Math.abs(value) + 0.5 + 1e-9);
  return value < 0 ? -rounded : rounded;
}

export function formatCents(cents: number): string {
  const sign = cents < 0 ? "-" : "";
  const abs = Math.abs(cents);
  return `${sign}${Math.floor(abs / 100)}.${String(abs % 100).padStart(2, "0")}`;
}

File: .lmbrain-lite/batteria/soluzioni/py-ledger/ledger/money.py
"""Currency conversion of integer cents."""

from decimal import ROUND_HALF_UP, Decimal

from . import rates


class UnknownCurrency(ValueError):
    """A currency code that the rate table does not know."""


def convert(cents, frm, to):
    """Convert `cents` of currency `frm` into cents of currency `to`.

    The exact result is rounded to the nearest cent, halves away from zero
    (50 GBP cents are 58.5 EUR cents, which becomes 59; -58.5 becomes -59).
    An unknown currency code raises UnknownCurrency.
    """
    for code in (frm, to):
        if code not in rates.RATES_TO_EUR:
            raise UnknownCurrency(code)
    if frm == to:
        return cents
    exact = Decimal(cents) * rates.rate(frm, to)
    return int(exact.quantize(Decimal(1), rounding=ROUND_HALF_UP))
